src/model/presentation/xml/SVGFile.java

```diff
@@ -93,9 +120,21 @@ public void addShapes(List<Shape> shapes) {
 		if (shapes != null)
 			this.shapes.addAll(shapes);
 	}
+	
+	@Override
+	public void insert(Presentation svg) {
+		if(this==svg){
+			try {
+				throw new Exception();
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+			this.shapes.add(svg);
+	}
 
 	@Override
-	public void display() {
+	public String display() {
 		this.xmlcontent = "";
```

Comment:
I don't like the name "display" because the method actually retrieves all the xml of the shapes and returns the xml that will be in the svg file. So it does not display anything, it constructs or retrieves xml content so maybe something like "createXMLContent" ?